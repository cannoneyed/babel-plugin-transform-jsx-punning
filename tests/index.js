import * as babel from 'babel-core';
import plugin from '../src';

const canonicalExample = `
<someTag someAttribute someOtherAttribute={hey} />
`;

it('matches snapshot', () => {
  const {code} = babel.transform(canonicalExample, {plugins: [plugin]});
  expect(code).toMatchSnapshot();
});

it('transforms boolean attributes with no value', () => {
  const input = '<someTag someAttribute />;'
  const output = '<someTag someAttribute={someAttribute} />;'
  const {code} = babel.transform(input, {plugins: [plugin]});
  expect(code).toEqual(output);
});

it('transforms boolean attributes with no value, leaving those with values alone', () => {
  const input = '<someTag someAttribute somethingElse={10} />;'
  const output = '<someTag someAttribute={someAttribute} somethingElse={10} />;'
  const {code} = babel.transform(input, {plugins: [plugin]});
  expect(code).toEqual(output);
});

it('doesn\'t transform attributes with values', () => {
  const input = '<someTag someAttribute={something} />;'
  const output = '<someTag someAttribute={something} />;'
  const {code} = babel.transform(input, {plugins: [plugin]});
  expect(code).toEqual(output);
});

it('doesn\'t transform attributes with boolean values', () => {
  const input = '<someTag someAttribute={false} />;'
  const output = '<someTag someAttribute={false} />;'
  const {code} = babel.transform(input, {plugins: [plugin]});
  expect(code).toEqual(output);
});