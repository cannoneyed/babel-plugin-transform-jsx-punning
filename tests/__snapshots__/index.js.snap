// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
"
<someTag someAttribute={someAttribute} someOtherAttribute={hey} />;"
`;
